### ISSUE TITLE
React application renders twice per page load after applying SC2020-003-435698 fixes

### ISSUE THREAD

**Issue by jimmah** [Sitecore/jss#462]
## Description
After applying the [required changes](https://github.com/Sitecore/jss/commit/2a2b87e31825b583cf27ea210731decf22715c75) for SC2020-003-435698 to our integrated mode JSS 13 application, the application appears to be being rendered twice server-side then twice client-side. 

To check what's going on:
- I've added `console.log('MyApp->server->renderView');` to `renderView` in `server/server.js`
- I've added the following log statements to `render` in `src/RouteHandler.js`:
  - `console.log('MyApp->RouteHandler->render');`
  - `console.log('IS_SERVER:', isServer());`

With the changes applied, I get the following log output in the server logs for each page load:
````
MyApp->server-renderView
MyApp->RouteHandler->render
IS_SERVER: true
MyApp->server-renderView
MyApp->RouteHandler->render
IS_SERVER: true
````

And the following in the browser console for each page load:
````
MyApp->RouteHandler->render
IS_SERVER: false
MyApp->RouteHandler->render
IS_SERVER: false
````

## Expected behavior
The application should be rendered once per page load.

With the changes reverted, I get the following log output in the server logs for each page load:
````
MyApp->server-renderView
MyApp->RouteHandler->render
IS_SERVER: true
````

And the following in the browser console for each page load:
````
MyApp->RouteHandler->render
IS_SERVER: false
````

## Steps To Reproduce
1. Create a default sample react app using `jss create some-app react`
2. Add the log statements above.
3. Follow the required steps to deploy it to Sitecore for integrated mode usage.
4. Load the app in the browser.
5. Check the log files to see how many times the application is rendered.

## Possible Fix
Not sure, but reverting the changes fixes the problem.

## Your Environment
* Sitecore Version: 9.3.0
* JSS Version: 13.2.2
* Browser Name and version: Chrome 85.0.4183.121
* Operating System and version (desktop or mobile): Windows 10 Pro
* Link to your project (if available): Internal client project, unable to share.

## Screenshots
![image](https://user-images.githubusercontent.com/587226/94913192-b9001180-04a0-11eb-92b4-69a229adf253.png)
![image](https://user-images.githubusercontent.com/587226/94913274-e056de80-04a0-11eb-9e30-45ccfb5d0e38.png)

**Comment by nickwesselman** [comment]
@jimmah The client-side 2X render call is expected due to a change in `ssrRenderComplete`. React should handle this efficiently.
![image](https://user-images.githubusercontent.com/27013789/95769327-b615d580-0c85-11eb-8c29-6a068b804805.png)

However we are unable to replicate the 2X call to `renderView` on JSS 13 w/ integrated mode. It's very odd that these changes would have any effect there, since this method is invoked external to the application itself. I'd recommend adding some additional debug information to your log statements in `renderView` to determine the source. The `path` argument perhaps?

**Comment by jimmah** [comment]
Hi @nickwesselman, thanks for investigating.

Logging the `path` has indeed pointed at the issue -- it's actually the 404 in the console in the screenshot I attached that is causing the second render call. That's a bug in one of our components we've already got on the list to fix. As long as the 2x client-side render is expected, we can close this then.

Apologies for the time waster and thanks for your help!
